db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/StatementSchemaPermission.java

```diff
@@ -186,6 +186,10 @@ protected boolean hasPermissionOnSchema(LanguageConnectionContext lcc,
 							Authorizer.PUBLIC_AUTHORIZATION_ID,
 							dbo);
 				}
+				if (rd == null && lcc.getCurrentGroupUser(activation) != null) {
```

Comment:
Similar to StatementTablePermission.java, we need to add the following before we check role permissions:
If NONE, whether currentGroupUser has schema level permission for the schema this database object belongs to.